spec/interfaces/protocols/length_protocol_spec.rb

```diff
@@ -46,6 +46,50 @@ def write(data); $buffer = data; end
     end
 
     describe "read" do
+      it "caches data for reads correctly" do
+        @interface.instance_variable_set(:@stream, LengthStream.new)
+        @interface.add_protocol(LengthProtocol, [
+          0, # bit offset
+          8, # bit size
+          0,  # length offset
+          1,  # bytes per count
+          'BIG_ENDIAN'], :READ_WRITE)
+        $buffer = "\x02\x03\x02\x05"
+        packet = @interface.read
+        expect(packet.buffer.length).to eql 2
+        expect(packet.buffer).to eql "\x02\x03"
+        packet = @interface.read
+        expect(packet.buffer.length).to eql 2
+        expect(packet.buffer).to eql "\x02\x05"
+        expect(@interface.read_protocols[0].read_data("\x03\x01\x02\x03\x04\x05")).to eql "\x03\x01\x02"
+        expect(@interface.read_protocols[0].read_data("")).to eql "\x03\x04\x05"
+        expect(@interface.read_protocols[0].read_data("")).to eql :STOP
+      end
+
+      it "caches data for reads correctly with multiple protocols" do
+        @interface.instance_variable_set(:@stream, LengthStream.new)
+        @interface.add_protocol(LengthProtocol, [
+          0, # bit offset
+          8, # bit size
+          0,  # length offset
+          1,  # bytes per count
+          'BIG_ENDIAN'], :READ_WRITE)
+        @interface.add_protocol(LengthProtocol, [
+          0, # bit offset
+          8, # bit size
+          0,  # length offset
+          1,  # bytes per count
+          'BIG_ENDIAN',
+          1], :READ_WRITE)
```

Comment:
This feels like a weird example to me. Adding two length protocols? The first doesn't have discard_leading_bytes set so it is 0 while the second has 1. I'm not sure how both result in a packet buffer of a single byte. Can you comment this example better.